.github/workflows/release.yml

```diff
@@ -42,6 +42,38 @@ jobs:
           path: ./dist/*.tar.gz
           retention-days: 5
 
+  test-sdist:
+    name: Test sdist
+    needs: [build-sdist]
+    runs-on: ubuntu-latest
+    container:
+      image: "osgeo/gdal:ubuntu-small-3.4.0" # python 3.8.10
+
+    steps:
+      - name: Download sdist from artifacts
+        uses: actions/download-artifact@v2
+        with:
+          path: wheelhouse
+
+      - name: Install python
+        run: |
+          apt-get update && apt-get install -y python3 python3-pip
+
+      - name: Build from sdist
+        shell: bash
+        run: |
+          python3 -m pip install --no-cache-dir cython~=0.29
+          python3 -m pip install --no-cache-dir --no-build-isolation wheelhouse/artifact/*.tar.gz
+          python3 -m pip list
+
+      - name: Run tests
+        shell: bash
+        run: |
+          cd ..
+          python3 -m pip install --no-cache-dir pytest pandas pyproj shapely
+          python3 -m pip install --no-cache-dir --no-deps geopandas
```

Comment:
Maybe move those two install lines to the step above, so those are included in the `pip list` output?